simulations/vip-442/bscmainnet.ts

```diff
@@ -0,0 +1,118 @@
+import { expect } from "chai";
+import { BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+import { NETWORK_ADDRESSES } from "src/networkAddresses";
+import { initMainnetUser, setRedstonePrice } from "src/utils";
+import { checkRiskParameters } from "src/vip-framework/checks/checkRiskParameters";
+import { checkVToken } from "src/vip-framework/checks/checkVToken";
+import { checkInterestRate } from "src/vip-framework/checks/interestRateModel";
+import { forking, testVip } from "src/vip-framework/index";
+
+import vip442, {
+  CONVERSION_INCENTIVE,
+  PROTOCOL_SHARE_RESERVE,
+  SOL_CHAINLINK_FEED,
+  SOL_INCENTIVES,
+  VANGUARD_TREASURY,
+  converterBaseAssets,
+  marketSpec,
+} from "../../vips/vip-442/bscmainnet";
+import ERC20_ABI from "./abi/ERC20.json";
+import COMPTROLLER_ABI from "./abi/LegacyPoolComptroller.json";
+import VTOKEN_ABI from "./abi/LegacyPoolVToken.json";
+import RESILIENT_ORACLE_ABI from "./abi/ResilientOracle.json";
+import SINGLE_TOKEN_CONVERTER_ABI from "./abi/SingleTokenConverter.json";
+
+const BLOCKS_PER_YEAR = BigNumber.from(10512000);
+
+const { RESILIENT_ORACLE, ACCESS_CONTROL_MANAGER, NORMAL_TIMELOCK, CHAINLINK_ORACLE, VTREASURY } =
+  NETWORK_ADDRESSES.bscmainnet;
+
+const MAINNET_USER = "0x4Ed6Cf63bd9C009d247ee51224Fc1c7041f517F1";
+const SOL_PRICE = parseUnits("200.65", 18);
+
+forking(46332892, async () => {
+  let theVanguardTreasuryBalancePrev: BigNumber;
+  let solHolder: any;
+  const resilientOracle = new ethers.Contract(RESILIENT_ORACLE, RESILIENT_ORACLE_ABI, ethers.provider);
+  const vToken = new ethers.Contract(marketSpec.vToken.address, VTOKEN_ABI, ethers.provider);
+  const comptroller = new ethers.Contract(marketSpec.vToken.comptroller, COMPTROLLER_ABI, ethers.provider);
+  const sol = new ethers.Contract(marketSpec.vToken.underlying.address, ERC20_ABI, ethers.provider);
+
+  before(async () => {
+    await setRedstonePrice(CHAINLINK_ORACLE, marketSpec.vToken.underlying.address, SOL_CHAINLINK_FEED, NORMAL_TIMELOCK);
```

Comment:
We don't use Redstone

Reply:
https://github.com/VenusProtocol/vips/pull/474/commits/4997aaee6d2f4433ef205c49fb6761f9bd951cec

We have to set the stale period for chainlink oracle to one year.